src/plugins/telemetryTable/collections/TableRowCollection.js

```diff
@@ -92,59 +89,86 @@ define(
                 }
             }
 
-            sortedLastIndex(rows, testRow) {
-                return this.sortedIndex(rows, testRow, _.sortedLastIndex);
-            }
+            sortAndMergeRows(rows) {
+                const sortedRowsToAdd = this.sortCollection(rows);
 
-            /**
-             * Finds the correct insertion point for the given row.
-             * Leverages lodash's `sortedIndex` function which implements a binary search.
-             * @private
-             */
-            sortedIndex(rows, testRow, lodashFunction = _.sortedIndexBy) {
                 if (this.rows.length === 0) {
```

Comment:
This feels like an unexpected side-effect of the function, and I know it already existed in the code. In the interests of time let's file a followup to fix it. The calling code should be explicitly resetting the TableRowCollection.